src/app/portlets/dot-edit-page/layout/dot-edit-layout/dot-edit-layout.component.ts

```diff
@@ -14,25 +17,36 @@ import { Observable } from 'rxjs/Observable';
 export class DotEditLayoutComponent implements OnInit {
     @ViewChild('editLayoutGrid') editLayoutGrid: DotEditLayoutGridComponent;
 
-    pageView: Observable<DotPageView>;
+    grid: Observable<DotLayoutGridBox[]>;
+    page: DotPageView;
 
-    constructor(public router: Router, private route: ActivatedRoute, private pageViewService: PageViewService) {}
+    constructor(
+        private dotEditLayoutService: DotEditLayoutService,
+        private pageViewService: PageViewService,
+        private route: ActivatedRoute,
+        public router: Router
+    ) {}
 
     ngOnInit(): void {
-        this.pageView = this.route.data.pluck('pageView');
+        this.grid = this.route.data
+            .pluck('pageView')
+            .do((pageView: DotPageView) => {
+                this.page = pageView;
+            })
+            .map((pageView: DotPageView) => {
```

Comment:
It wasn't 100% necessary to do ```this.grid``` an Observable but because I need to transform using ```.map``` the stream I rather to make it.